src/foam/lib/query/NegateParser.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.query;
+
+import foam.lib.parse.Alt;
+import foam.lib.parse.Literal;
+import foam.lib.parse.LiteralIC;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.Seq;
+import foam.mlang.predicate.Not;
+
+public class NegateParser extends foam.lib.parse.ProxyParser  {
+
+  public NegateParser(Parser exprParser) {
+    setDelegate(new Alt(new Seq(new Literal("-"),

```

Comment:
Simpler to put the Alt in the sequence, and use a Seq1 so you don't need the Object[] dance inside.

```
setDelegate(new Seq1(1,
  new Alt(new Literal("-"), new LiteralIC("not "),
  exprParser));
```

Then in parse do

```
setArg1((foam.mlang.Expr)ps.value());
```